planning/scenario_planning/lane_driving/behavior_planning/turn_signal_decider/src/data_manager.cpp

```diff
@@ -48,56 +48,68 @@ lanelet::ConstLanelets pathToLanes(
 
 namespace turn_signal_decider
 {
-DataManager::DataManager()
-: is_map_ready_(false), is_path_ready_(false), is_pose_ready_(false), tf_listener_(tf_buffer_)
+DataManager::DataManager(rclcpp::Node::SharedPtr node)
+: is_map_ready_(false), is_path_ready_(false), is_pose_ready_(false), node_(node)
 {
+  if (auto node_ptr = node_.lock()) {
+    tf_buffer_ = std::make_shared<tf2_ros::Buffer>(node_ptr->get_clock());
+    tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
+  } else {
+    throw std::runtime_error("Node no longer valid");
+  }
 }
 
-void DataManager::onPathWithLaneId(const PathWithLaneId & msg)
+void DataManager::onPathWithLaneId(PathWithLaneId::SharedPtr msg)
 {
-  path_ = msg;
+  path_ = *msg;
   is_path_ready_ = true;
   if (is_map_ready_) {
     path_lanes_ = pathToLanes(path_, lanelet_map_ptr_);
   }
 }
 
-void DataManager::onLaneletMap(const autoware_lanelet2_msgs::MapBin & map_msg)
+void DataManager::onLaneletMap(autoware_lanelet2_msgs::msg::MapBin::SharedPtr map_msg)
 {
   lanelet_map_ptr_ = std::make_shared<lanelet::LaneletMap>();
   lanelet::utils::conversion::fromBinMsg(
-    map_msg, lanelet_map_ptr_, &traffic_rules_ptr_, &routing_graph_ptr_);
+    *map_msg, lanelet_map_ptr_, &traffic_rules_ptr_, &routing_graph_ptr_);
   is_map_ready_ = true;
 
   if (is_path_ready_) {
     path_lanes_ = pathToLanes(path_, lanelet_map_ptr_);
   }
 }
 
-void DataManager::onVehiclePoseUpdate(const ros::TimerEvent & event)
+void DataManager::onVehiclePoseUpdate()
 {
-  try {
-    const auto current_time = ros::Time::now();
-    const auto transform =
-      tf_buffer_.lookupTransform("map", "base_link", current_time, ros::Duration(0.1));
-    vehicle_pose_.pose.position.x = transform.transform.translation.x;
-    vehicle_pose_.pose.position.y = transform.transform.translation.y;
-    vehicle_pose_.pose.position.z = transform.transform.translation.z;
-    vehicle_pose_.pose.orientation.x = transform.transform.rotation.x;
-    vehicle_pose_.pose.orientation.y = transform.transform.rotation.y;
-    vehicle_pose_.pose.orientation.z = transform.transform.rotation.z;
-    vehicle_pose_.pose.orientation.w = transform.transform.rotation.w;
-    vehicle_pose_.header.frame_id = "map";
-    vehicle_pose_.header.stamp = current_time;
-    is_pose_ready_ = true;
-  } catch (tf2::TransformException & ex) {
-    // if pose has never arrived before, then wait for localizatioon
-    if (!is_pose_ready_) {
-      ROS_WARN_STREAM_THROTTLE(5, ex.what());
-    } else  // if tf suddenly stops comming, then there must be something wrong.
-    {
-      ROS_ERROR_STREAM_THROTTLE(5, ex.what());
+  if (auto node_ptr = node_.lock()) {
+    try {
+      const auto current_time = node_ptr->now();
+      const auto transform =
+        tf_buffer_->lookupTransform("map", "base_link", current_time, rclcpp::Duration(1.0));
```

Comment:
It seems like it was originally 0.1 seconds for duration.
Is this change intentional?
Also, do you know which constructor is called from double? It looks like most of the constructor is expecting nanoseconds. 
http://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1Duration.html#af2ab74e1b3f70436bb0451dd8a17c8e8

It might be more explicit if we use rclcpp::Duration::from_seconds() function

Reply:
Good catch, I copied that Duration from elsewhere and forgot to update the value, thanks.

Fixed in 7270016